client.cpp

```diff
@@ -64,68 +64,71 @@ void Client::SendMessage(const char* message) {
     }
 }
 
-void Client::SendMessageThread() {
+void Client::sendMessageThread() {
     std::string message;
     while (true) {
         std::getline(std::cin, message);
         if (message == "exit") {
             break;
         }
-        if (sizeof(message) > UINT_MAX) {
-            std::cout << "message too long\n";
+        if (message.length() > UINT_MAX) {
+            std::cout << "Message too long\n";
         }
         else {
-            SendMessage(message.c_str());
+            sendMessage(message.c_str());
         }
     }
 }
 
+uint32_t Client::parseSize(const char* buffer) {
```

Comment:
```suggestion
std::uint32_t Client::parseSize(const std::string& buffer) {
```
and convert to string before sending